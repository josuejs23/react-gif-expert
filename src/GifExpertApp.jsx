import { useState } from "react"
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ()=>{
    
    const [categories, setCategories] = useState([])

    const onAddCategory = (newCategory)=>{
        if(categories.includes(newCategory)) return;
        console.log(newCategory)
        setCategories([newCategory, ...categories]);
    }

    console.log(categories)
    return(
        <>
            <h1>Gif-Expert-App</h1>

            <AddCategory 
                onNewCategory={ (category)=>onAddCategory(category)}
            />
            
            {
                categories.map( (categorie )=> (
                    <GifGrid category={categorie} key={categorie}/>
                )
                )
            }
        </>
    )
}