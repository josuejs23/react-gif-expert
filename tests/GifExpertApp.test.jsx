import { render } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('TEST on <GifExpertApp/>', ()=>{
    test('Should match with the snapshot',()=>{
        const {container}= render(<GifExpertApp/>)
        expect(container).toMatchSnapshot();

    })
})