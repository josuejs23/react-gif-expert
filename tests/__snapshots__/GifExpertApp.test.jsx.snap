// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST on <GifExpertApp/> Should match with the snapshot 1`] = `
<div>
  <h1>
    Gif-Expert-App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="search category.."
      type="text"
      value=""
    />
  </form>
</div>
`;
